LEAF_Request_Portal/sources/System.php

```diff
@@ -1016,4 +1016,97 @@ public function getHistory($filterById)
     {
         return $this->dataActionLogger->getHistory($filterById, null, \LoggableTypes::PRIMARY_ADMIN);
     }
+
+    /**
+     * Returns every customization in the following categories:
+     * 1.) template editor
+     * 2.) indictor html field (form editor)
+     * 3.) indicator html print field (form editor)
+     * 4.) email editor
+     *
+     * @return array array with site customizations broken up into four sub arrays: 
+     * 'templateEditor', 'formIndicatorHtml', 'formIndicatorHtmlPrint', 'emailEditor'
+     */
+    public function getCustomizations()
+    {
+        $templateEditor = array();
+        $formIndicatorHtml = array();
+        $formIndicatorHtmlPrint = array();
+        $emailEditor = array();
+
+        $result = [
+            'templateEditor' => $this->getTemplateEditorCustomizations(),
+            'formIndicatorHtml' => $this->getFormIndicatorHtmlCustomizations(),
+            'formIndicatorHtmlPrint' => $this->getFormIndicatorHtmlPrintCustomizations(),
+            'emailEditor' => $this->getEmailEditorCustomizations(),
+        ];
+
+        return $result;
+    }
+
+    /**
+     * Returns array of template editor customization descriptions.
+     *
+     * @return array array of template editor customization descriptions
+     */
+    private function getTemplateEditorCustomizations($subfolder = null)
+    {
+        global $config;
+        $result = array();
+
+        if(isset($subfolder))
+        {
+            //make sure there's a trailing slash and no leading slash
+            $subfolder = ltrim(rtrim($subfolder,"/"), "/") . "/";
+        }
+        $cleanPortalPath = str_replace("/", "_", $config->portalPath);
+        $customTemplatePath = __DIR__ . "/../templates/" . $subfolder . "custom_override/";
```

Comment:
on ```$this->getTemplateEditorCustomizations()``` when ```$subfolder``` is null, will this still work? and resolve to /../templates/custom_override/ instead of /../templates/null/custom_override/

Reply:
NULL is always converted to an empty string, see https://www.php.net/manual/en/language.types.string.php#language.types.string.casting